File: backend/test_api.js
const fetch = require('node-fetch');
async function run() {
  // Login first to get token
  let res = await fetch('http://127.0.0.1:8000/api/v1/auth/login', {
    method: 'POST',
    headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
    body: 'username=admin%40vendorbridge.com&password=AdminPassword123!'
  });
  let data = await res.json();
  let token = data.access_token;
  
  // Try to add user
  let addRes = await fetch('http://127.0.0.1:8000/api/v1/users/', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${token}` },
    body: JSON.stringify({
      first_name: 'Test',
      last_name: 'User',
      email: 'test4@test.com',
      phone: '',
      role_id: '123e4567-e89b-12d3-a456-426614174000', // random uuid
      password: 'password123',
      is_active: true
    })
  });
  console.log(await addRes.text());
}
run();

File: backend/test_api.py
import requests
from backend.app.core.security import create_access_token
from backend.app.core.database import SessionLocal
from backend.app.models.user import User
from backend.app.models.role import Role

db = SessionLocal()
admin = db.query(User).join(Role).filter(Role.name == "Admin").first()
dummy = db.query(User).filter(User.email == "dummy@example.com").first()

token = create_access_token(admin.id)

print("Sending PATCH request...")
res = requests.patch(
    f"http://127.0.0.1:8000/api/users/{dummy.id}",
    json={"password": "newpassword123", "role_id": str(dummy.role_id), "is_active": True},
    headers={"Authorization": f"Bearer {token}"}
)
print("Status:", res.status_code)
print("Response:", res.text)
